Src/BootCamp.Chapter/Shop.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Linq.Expressions;
+
+namespace BootCamp.Chapter
+{
+    public class Shop
+    {
+        private decimal _money;
+        public decimal Money { get { return _money; } }
+        private ShopInventory _inventory;
+        public Item[] Items { get { return _inventory.Items; } }
+
+        public Shop(decimal money)
+        {
+            _money = money;
+            _inventory = new ShopInventory();
+        }
+
+        /// <summary>
+        /// Adds item to the stock.
+        /// If item of same name exists, does nothing.
+        /// </summary>
+        public void Add(Item item)
+        {
+            try
+            {
+                _inventory.AddItem(item ?? throw new ArgumentNullException("Item is Null"));
+            }
+            catch (InventoryIsFullException msg)
+            {
+                Console.WriteLine(msg);
+            }
+        }
+
+        /// <summary>
+        /// Removes item from the stock.
+        /// If item doesn't exist, does nothing.
+        /// </summary>
+        /// <param name="name"></param>
+        public void Remove(string name)
+        {
+            _inventory.RemoveItem(name ?? throw new ArgumentNullException("Item is Null"));
+        }
+
+        /// <summary>
+        /// Player can sell items to a shop.
+        /// All items can be sold.
+        /// Shop looses money.
+        /// </summary>
+        /// <returns>Price of an item.</returns>
+        public decimal Buy(Item item)
+        {
+            if(item != null)
```

Comment:
Validate null first, execute logic after. Else is not needed, becuase throwing an exception exists the function.

Reply:
Agreed. Utilised the discard (_) keyword or symbol to do a check to seew whether 'Item' was Null through implementing the Null-Coalescing function.